### PR TITLE
Fix: Internal `BaseTypography` removed

### PR DESCRIPTION
**Pull request by Jialecl**
**Checklist**

- [x] The Build process is done without errors. All tests pass in the `/lib` directory.
- [x] Self-reviewed the code before submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed.
- [x] Added/updated tests as needed.

**Description**

Removing the internal `BaseTypography` component to prevent build issues with some apps using a different bundler. The properties set using the Typography context where no applied correctly.

We can revisit this in the future since it is not our fault or an implementation issue but, for now, we will solve it like this.

### CODE REVIEW

File: packages/lib/src/nav-tabs/Tab.tsx [hunk 1]
```diff
@@ -60,7 +59,7 @@ const DxcTab = forwardRef(
             </TabIconContainer>
           )}
           <DxcFlex alignItems="center" gap="0.5rem">
-            <BaseTypography
+            <LabelContainer
```
Comment (GomezIvann) [inline packages/lib/src/nav-tabs/Tab.tsx:62]: I don't understand this implementation. Why aren't you consuming the theme from the provider as in the other styled-components?

I know it is an internal component without a `ThemeProvider` but it should only be used outside the NavTabs, which do have such a provider, so this component would have access to it.

File: packages/lib/src/nav-tabs/Tab.tsx [hunk 2]
```diff
@@ -137,6 +133,28 @@ const Tab = styled.a<{
...
 
+const LabelContainer = styled.span<{
+  color: string;
+  fontFamily: string;
+  fontSize: string;
+  fontStyle: string;
+  fontWeight: string;
+}>`
+  display: inline;
+  color: ${(props) => props.color};
```
Comment (GomezIvann) [inline packages/lib/src/nav-tabs/Tab.tsx:144]: Use `props.theme.xxx` instead and remove the props.

File: packages/lib/src/tabs/Tab.tsx [hunk 3]
```diff
@@ -111,12 +110,9 @@ const DxcTab = forwardRef(
               fontSize={colorsTheme.tabs.fontSize}
```
Comment (GomezIvann) [inline packages/lib/src/tabs/Tab.tsx:110]: Same as in the Nav Tabs, don't use props and instead, consume the theme.

File: packages/lib/src/tabs/TabLegacy.tsx [hunk 4]
```diff
@@ -170,6 +166,28 @@ const MainLabelContainer = styled.div<{
       : "unset"};
 `;
 
+const LabelContainer = styled.span<{
+  color: string;
```
Comment (GomezIvann) [inline packages/lib/src/tabs/TabLegacy.tsx:170]: Same as in the Nav Tabs, don't use props and instead, consume the theme.

File: packages/lib/src/paragraph/Paragraph.tsx [hunk 5]
```diff
@@ -1,20 +1,35 @@
...
 import useTheme from "../useTheme";
-import BaseTypography from "../utils/BaseTypography";
 
 const DxcParagraph = ({ children }: { children: React.ReactNode }): JSX.Element => {
   const colorsTheme = useTheme();
 
   return (
-    <BaseTypography
-      as="p"
+    <ParagraphContainer
```
Comment (GomezIvann) [inline packages/lib/src/paragraph/Paragraph.tsx:8]: Implement the component as we usually do Jiale, use the ThemeProvider!

`<ThemeProvider theme={colorsTheme.paragraph}>`

**Review by GomezIvann — CHANGES_REQUESTED** [review]

File: packages/lib/src/typography/Typography.tsx [hunk 6]
```diff
@@ -1,14 +1,120 @@
...
-    textOverflow={textOverflow}
-    whiteSpace={whiteSpace == null && textOverflow != null && textOverflow !== "unset" ? "nowrap" : whiteSpace}
-    {...props}
-  >
-    {children}
-  </BaseTypography>
-);
+const TypographyContext = React.createContext<TypographyContextPropTypes | null>(null);
 
+const ValidTypographyTags = [
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply packages/lib/src/typography/Typography.tsx]: I did the change, we need this fix. We can discuss it later.

File: packages/lib/src/typography/Typography.tsx [hunk 7]
```diff
@@ -1,14 +1,120 @@
...
+  "h5",
+  "h6",
+  "p",
+  "pre",
+  "small",
+  "span",
+  "strong",
+];
+
+const isValidTypography = (tag: keyof HTMLElementTagNameMap) => {
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply packages/lib/src/typography/Typography.tsx]: I did the change, we need this fix. We can discuss it later.

**Review by Mil4n0r — APPROVED** [review]

**Review by GomezIvann — APPROVED** [review]

**Pull request merged**